R-package/tests/testthat/test_basic.R

```diff
@@ -2079,7 +2079,8 @@ test_that("lgb.train() works with linear learners when Dataset has categorical f
     , metric = "mse"
     , seed = 0L
     , num_leaves = 2L
-    , categorical_feature = 1L
+    # indices in categorical feature start from 1L, so here should be 2L
+    , categorical_feature = 2L
```

Comment:
similar to my previous comment...is this absolutely necessary for this PR? Or could it be separate?

Reply:
Resolved via fab88db. Remove this change. And will open a separate issue for this.